src/ember-app.js

```diff
@@ -190,16 +190,28 @@ class EmberApp {
    * @param {string} path the URL path to render, like `/photos/1`
    * @param {Object} options
    * @param {string} [options.html] the HTML document to insert the rendered app into
+   * @param {metaData} [options.metaData] Per request specific data used in the app.
+   * @param {Boolean} [options.shouldRender] whether the app should do rendering or not. If set to false, it puts the app in routing-only.
+   * @param {Boolean} [options.disableShoebox] whether we should send the API data in the shoebox. If set to false, it will not send the API data used for rendering the app on server side in the index.html.
+   * @param {Integer} [options.destroyAppInstanceInMs] whether to destroy the instance in the given number of ms. This is a failure mechanism to not wedge the Node process (See: https://github.com/ember-fastboot/fastboot/issues/90)
    * @param {ClientRequest}
+   * @param {ClientResponse}
    * @returns {Promise<Result>} result
    */
   visit(path, options) {
     let req = options.request;
     let res = options.response;
+    let metaData = options.metaData;
     let html = options.html || this.html;
+    let disableShoebox = options.disableShoebox || false;
+    let destroyAppInstanceInMs = options.destroyAppInstanceInMs;
+    let hostWhitelist = this.hostWhitelist;
 
-    let bootOptions = buildBootOptions();
-    let fastbootInfo = new FastBootInfo(req, res, this.hostWhitelist);
+    let shouldRender = (options.shouldRender !== undefined) ? options.shouldRender : true;
```

Comment:
Does `options.hasOwnProperty('shouldRender') ? options.shouldRender : true;` work better (I am unsure if it is better...)?